### PR TITLE
Attention tweaks

### PR DESCRIPTION
**Pull request by PierreBoyeau**

### CODE REVIEW

File: src/scvi_v2/_module.py [hunk 1]
```diff
@@ -458,7 +464,7 @@ def loss(
         if self.qz_nn_flavor != "linear":
             qeps = inference_outputs["qeps"]
             eps = inference_outputs["z"] - inference_outputs["z_base"]
-            peps = dist.Normal(0, self.z_u_prior_scale)
+            peps = dist.Normal(0, jnp.exp(self.pz_scale))
```
Comment (adamgayoso) [inline src/scvi_v2/_module.py]: let's use softplus for all Normal scales?

File: src/scvi_v2/_components.py [hunk 2]
```diff
@@ -136,6 +136,59 @@ def __call__(self, x: NdArray, condition: NdArray, training: Optional[bool] = No
...
+            eps = jnp.reshape(eps, (eps.shape[0], eps.shape[1], eps.shape[2] * eps.shape[3]))
+
+        eps_ = MLP(
+            n_out=self.outerprod_dim,
+            n_hidden=self.n_hidden_mlp,
+            training=training,
+            activation=self.activation,
+        )(inputs=eps)
+        inputs = jnp.concatenate([query_embed, eps_], axis=-1)
+        residual = MLP(
```
Comment (canergen) [inline src/scvi_v2/_components.py:184]: It's slightly unclear whether the gradient should also stop for this MLP. It would be p(u|z) is independent of sample embedding the other thing would be p(u|eps) is independent of sample embedding.

File: src/scvi_v2/_module.py [hunk 3]
```diff
@@ -32,7 +32,7 @@ class _DecoderZX(nn.Module):
     n_out: int
     n_batch: int
     n_hidden: int = 128
-    activation: Callable = nn.softmax
+    activation: Callable = nn.softplus
```
Comment (adamgayoso) [inline src/scvi_v2/_module.py]: oh didn't mean here but whatever you want here!

File: src/scvi_v2/_components.py (hunk 2 above)
Comment (adamgayoso) [reply src/scvi_v2/_components.py:184]: my though here was that at this point in the flow, it becomes equivalent to like how the vanilla scvi decoder works except you have a featurized batch embedding

I'm not sure this line (181) is completely necessary or not

Comment (PierreBoyeau) [reply src/scvi_v2/_components.py:184]: Following up on this, I think the gradient stopping that occurs upstream (respectively in the encoder and decoder), should instead apply here.
In particular, we should stop the gradient L160, but not L181.
I will try to see if this improves integration, and update when I have results.

File: src/scvi_v2/_module.py [hunk 4]
```diff
@@ -465,8 +469,11 @@ def loss(
         if self.qz_nn_flavor != "linear":
             qeps = inference_outputs["qeps"]
             eps = inference_outputs["z"] - inference_outputs["z_base"]
-            peps = dist.Normal(0, jnp.exp(self.pz_scale))
-            kl_z = dist.kl_divergence(qeps, peps).sum(-1) if qeps is not None else -peps.log_prob(eps).sum(-1)
+            if self.z_u_prior:
+                peps = dist.Normal(0, jnp.exp(self.pz_scale))
```
Comment (justjhong) [inline src/scvi_v2/_module.py:494]: why is this exponentiated? Doesn't make sense for manually chosen scale

Comment (canergen) [reply src/scvi_v2/_module.py:494]: it's learnable (as optional argument) so we need to ensure it's positive.
